source/code/plugins/oms_common.rb

```diff
@@ -674,9 +674,18 @@ def create_ods_http(ods_uri, proxy={})
       #   serializer: method. serializer of the record
       # returns:
       #   HTTPRequest. request to ODS
-      def create_ods_request(path, record, compress, extra_headers=nil, serializer=method(:parse_json_record_encoding))
+      def create_ods_request(path, omsadmin_conf_path, record, compress, extra_headers=nil, serializer=method(:parse_json_record_encoding))
         headers = extra_headers.nil? ? {} : extra_headers
 
+        File.open(omsadmin_conf_path).each_line do |line|
```

Comment:
we shouldn't read the conf file for every request .. instead have it static in memory and read the file only if it is not initialized (first time after service is started/restarted)